R/generateHypotheses.R

```diff
@@ -1,192 +1,305 @@
-  binaries_sort <- lapply(binaries,FUN=sort) # sort alphanumerically to make it possible to remove duplicates at the end
-  binaries_sort_table <- data.table(t(as.data.table(binaries_sort)))
-  colnames(binaries_sort_table) <- c("a","b")
-  binary_interactions <- rbind(binary_interactions,binaries_sort_table)
-  binary_interactions
-}
 
-#' calculatePathlength
+#' Calculate pathlength from binary interactions
 #' @description Calculate shortest paths between all members of a binary interaction network.
 #' This function creates a table that contains all possible combinations of 2 from
-#' all input proteins (binomial coeffitient (length(proteins) over 2)).
+#' all input proteins (binomial coefficient (length(proteins) over 2)).
 #' @import data.table
-#' @import igraph
+#' @importFrom igraph graph_from_data_frame
+#' @importFrom igraph distances
 #' @param binaryInteractionTable data.table with binary interactions between "a" and "b" as returned from generateBinaryNetwork()
-#' @return data.table with three columns (x,y,dist) containing all possible combinations
-#' of proteins and their shortest distance as determined by the input binary interaction table
+#' @return data.table with three columns containing all possible combinations of proteins:
+#' \itemize{
+#'   \item x,y: Protein ids of the input proteins (all possible combinations)
+#'   \item dist: distance as determined by the input binary interaction table
+#' }
 #' @export
+#' @examples 
+#' ## Load example Data
+#' complexHypotheses <- exampleComplexHypotheses
+#' 
+#' ## Generate the binary network
+#' binaryInteractions <- generateBinaryNetwork(complex_hypotheses = complexHypotheses)
+#' 
+#' ## Calculate the path lengths
+#' shortestPaths <- calculatePathlength(binaryInteractions)
+#' 
+
 calculatePathlength <- function(binaryInteractionTable){
-  proteins <- unique(c(binaryInteractionTable$a,binaryInteractionTable$b)) # get all protein ids in interaction table
-  g <- graph_from_data_frame(binaryInteractionTable, directed = FALSE) # create undirected graph using the igraph package
-  distMatrix <- distances(g) #as.data.frame(shortest.paths(g, v=V(g), to=V(g),weights = NULL, mode="all")) # calculate shortest path between all vertices
+  ## get all protein ids in interaction table
+  proteins <- unique(c(binaryInteractionTable$a,binaryInteractionTable$b)) 
+  ## create undirected graph using the igraph package
+  g <- graph_from_data_frame(binaryInteractionTable, directed = FALSE)
+  ## calculate shortest path between all vertices
+  distMatrix <- distances(g) 
+  ## Output as a long format data.table
   combinations <- melt(distMatrix)
   combinations <- as.data.table(combinations)
   names(combinations) = c("x","y","dist")
   combinations$x <- as.character(combinations$x)
   combinations$y <- as.character(combinations$y)
-  combinations[]
+  
+  return(combinations)
```

Comment:
So what I get from that post is that it really doesn't matter and people do it both ways.
I find it a bit nicer to read when it is always there (because sometimes it is really neccessary (e.g. in ifs)

But this is your call. Let me know how you decide, we should the put it in the guidelines